tools/autograd/templates/VariableType.cpp

```diff
@@ -41,6 +43,52 @@ static void setattr(jit::Node* n, jit::Symbol name, double v)              { n->
 template<std::size_t N>
 static void setattr(jit::Node* n, jit::Symbol name, std::array<bool, N> v) { n->is_(name, std::vector<int64_t>(v.begin(), v.end())); }
 
+template<typename T>
+static jit::Value* createConstant(jit::Node* n, T value) {
+  return n->owningGraph()->createConstant(jit::as_tensor(value))->insertBefore(n)->output();
+}
+
+template<typename T>
+static void genericInsertInput(jit::Node* n, size_t idx, T value) {
+  n->insertInput(idx, createConstant(n, t));
+}
+
+void failPosAttr() {
+  throw std::runtime_error("unsupported type in setposattr. File a bug report!");
+}
+
+static void setposattr(jit::Node* n, size_t idx, const char *name, int64_t v)             { genericInsertInput(n, idx, v); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, const at::Scalar& v)   { genericInsertInput(n, idx, v); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, SparseTensor s)        { failPosAttr(); }
+static void setposattr(jit::Node* n, size_t idx, const char *name, const at::IntList& v)  {
+  using ArgumentStash = jit::tracer::ArgumentStash;
+  if (ArgumentStash::hasIntList(name)) {
+    auto info = ArgumentStash::popIntList(name);
+    for (size_t i = 0; i < info.size(); ++i) {
+      if (info[i] != nullptr) continue;
```

Comment:
When is `info[i]` null? When it is not a Variable?

Reply:
Exactly. Not all elements of that int list have to be traceable.